server/service/src/invoice/prescription/update/mod.rs

```diff
@@ -60,6 +60,23 @@ pub fn update_prescription(
     let invoice = ctx
         .connection
         .transaction_sync(|connection| {
+            // if the backdated_datetime is in the future, we assume we actually mean now
+            // This is a fix for where backdated_date is set to the end of the day even though the invoice is created in the morning
+            // TODO: backdated datetime Should possibly be a nullable update, and should be unset if it's in the future?
+            // https://github.com/msupply-foundation/open-msupply/issues/4979
+            let patch = match patch.backdated_datetime {
```

Comment:
Not entirely happy with this, the backdated_datetime needs to go into the Validate method, and also apply to the generate, so have put the logic here for now, hoping to clean up in 4979.

Possibly a frontend fix would be nicer and simpler?